Code/Framework/AzFramework/AzFramework/Physics/RigidBodyBus.h

```diff
@@ -79,13 +81,68 @@ namespace Physics
 
     using RigidBodyRequestBus = AZ::EBus<RigidBodyRequests>;
 
-
+    //! Notifications interface for a rigid body (static or dynamic).
     class RigidBodyNotifications
         : public AZ::ComponentBus
     {
+    private:
+        template<class Bus>
+        struct RigidBodyNotificationsConnectionPolicy : public AZ::EBusConnectionPolicy<Bus>
+        {
+            static void Connect(
+                typename Bus::BusPtr& busPtr,
+                typename Bus::Context& context,
+                typename Bus::HandlerNode& handler,
+                typename Bus::Context::ConnectLockGuard& connectLock,
+                const typename Bus::BusIdType& id = 0)
+            {
+                AZ::EBusConnectionPolicy<Bus>::Connect(busPtr, context, handler, connectLock, id);
+
+                AZ::Entity* entity = nullptr;
+                EBUS_EVENT_RESULT(entity, AZ::ComponentApplicationBus, FindEntity, id);
```

Comment:
Please don't use the macro form for bus calls, I'm in the process of removing that from the whole engine.

Reply:
Yes, will change!